census_view.cpp

```diff
@@ -932,6 +936,844 @@ inline std::vector<std::string> const& CensusViewDataViewModel::all_headers() co
     return view_.case_parms()[0].member_names();
 }
 
+namespace
+{
+
+// Declare the functions here, but implement after the CensusViewTable
+// declaration to avoid the "use of undefined type" error.
+
+// Get the cell value from the table.
+
+any_member<Input> const& cell_at(wxGridTableBase const& table, int row, int col);
+
+// Get the raw value from the table.
+
+Input const& row_at(wxGridTableBase const& table, int row);
+
+// Get the column name from the table.
+
+std::string const& col_name(wxGridTableBase const& table, int col);
+
+// class RangeIntEditor
+
+class RangeIntEditor
+    :public wxGridCellNumberEditor
+{
+  public:
+    // Construct wxGridCellNumberEditor with fake limits to use
+    // the wxSpinCtrl. Real limits will be set in BeginEdit function.
+    RangeIntEditor()
+        :wxGridCellNumberEditor {0, 1}
+    {
+        LMI_ASSERT(HasRange());
```

Comment:
Why do we do this? If it's really needed, could you please explain it in a comment?